api/v1alpha1/nicdevice_types.go

```diff
@@ -20,22 +20,37 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type NicDeviceConfigurationSpec struct {
+	ResetToDefault bool `json:"resetToDefault,omitempty"`
+	// TODO add NicConfigurationTemplateSpec once #2 is merged
+}
 
 // NicDeviceSpec defines the desired state of NicDevice
 type NicDeviceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Configuration *NicDeviceConfigurationSpec `json:"configuration,omitempty"`
+}
+
+type NicDevicePortSpec struct {
+	Pci              string `json:"pci"`
```

Comment:
do we want to name it PCI or leave as is ?

Reply:
I don't have a preference, let's capitalize it